src/DynamoCore/Extensions/ExtensionManager.cs

```diff
@@ -8,6 +8,40 @@
 
 namespace Dynamo.Extensions
 {
+
+    /// <summary>
+    /// An object which may request extensions to be loaded and added to the extensionsManager
+    /// </summary>
+    public interface IExtensionSource
+    {
+        /// <summary>
+        /// event that is raised when package manager requests an extension be loaded
+        /// which was found within a package it is currently loading.
+        /// </summary>
+        event Func<string, IExtension> RequestLoadExtensionHandler;
+
+        /// <summary>
+        /// event that is raised when package manager requests an Extension to be added to
+        /// the list of currently loaded extensions. The Extension was found within a package that
+        /// is currently being loaded. This is usually raised directly after an Extension load occurs.
+        /// </summary>
+        event Action<IExtension> RequestAddExtensionHandler;
+
+
+        /// <summary>
+        /// event that is raised when package manager requests a ViewExtension be loaded
+        /// which was found within a package it is currently loading.
+        /// </summary>
+        event Func<string, dynamic> RequestLoadViewExtension;
+
+        /// <summary>
+        /// event that is raised when package manager requests a viewExtension to be added to
+        /// the list of currently loaded extensions. The ViewExtension was found within a package that
+        /// is currently being loaded. This is usually raised directly after a ViewExtension load occurs.
+        /// </summary>
+        event Action<dynamic> RequestAddViewExtension;
```

Comment:
All of these events have an unnecessarily large level of focus on the package manager.

Reply:
`ViewExtensions` aren't really part of `DynamoCore`. Shouldn't this be in the `DynamoCoreWpf` layer? It might seem duplicative, but I don't think this really results in that much extra code.

Reply:
indeed thats what I intended to do first, but I don't believe there is a View `packageLoader` analog where we would load extension manifests from package from (again) - since these are already loaded in the package manager extension (not a view extension) - we would not be able to reference an interface in `DynamoCoreWpf` without kind of entailing an extension with wpf references... so I decided to use dynamic here and add these events.

I could definitely be convinced to look for an alternative - any thoughts on that?